tests/test_reduction_ops.py

```diff
@@ -250,6 +251,22 @@ def test_accuracy_cumsum(shape, dtype):
     gems_assert_close(res_out, ref_out, dtype, reduce_dim=shape[dim])
 
 
+@pytest.mark.parametrize("shape", REDUCTION_SHAPES)
```

Comment:
I test shape=(1024 * 1024 * 64, 1) and get a wrong result. Can you try this shape in your environment?

Reply:
I got wrong result too.
The bug is located as below. **The root cause is the precision error due to float**. **inp** is **bool**, and it is converted to **float32** to calculate the cumsum. It should be **int32**. 
Same precision issue may happen when the input dtype is float16 or bfloat16. I will correct them together later.
```
# wrong
inp_vals = tl.load(inp_ptrs, mask=mask).to(tl.float32)
result = tl.cumsum(inp_vals, axis=0)

# correct
inp_vals = tl.load(inp_ptrs, mask=mask).to(tl.int32)
result = tl.cumsum(inp_vals, axis=0)
```

Reply:
done